solidity/contracts/BancorNetwork.sol

```diff
@@ -34,10 +39,20 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient, F
     uint256 private constant CONVERSION_FEE_RESOLUTION = 1000000;
     uint256 private constant AFFILIATE_FEE_RESOLUTION = 1000000;
 
+    struct ConversionStep {
+        IBancorConverter converter;
+        ISmartToken smartToken;
+        IERC20Token sourceToken;
+        IERC20Token targetToken;
+        uint256 minReturn;

```

Comment:
About that - we didn't discuss it yet cause I tried not to change it until we finish the initial review but I'm thinking to remove the minimum return completely from the converters and just check it in the network.
If you don't see an issue with it, it will make this point irrelevant since the value will be removed from the ConversionStep struct.

Reply:
@yudilevi:
I think we should keep it, because it allows us to remove the check of last-index in every iteration on the conversion path, which increases the overall gas consumption, as well as the compiler-stack usage as you may recall.

Reply:
It will allow us to remove it anyway because we'll just do the check in the network after we're done with the loop